graylog2-web-interface/src/components/permissions/EntityShareModal.jsx

```diff
@@ -25,18 +25,40 @@ const EntityShareModal = ({ description, entityId, entityType, entityTitle, onCl
   const { state: entityShareState } = useStore(EntityShareStore);
   const [disableSubmit, setDisableSubmit] = useState(false);
   const entityGRN = createGRN(entityType, entityId);
+  const granteesSelectRef = useRef();
 
   useEffect(() => {
     EntityShareActions.prepare(entityGRN);
   }, [entityGRN]);
 
   const _handleSave = () => {
     setDisableSubmit(true);
+    const granteesSelect = granteesSelectRef?.current;
+    const garnteesSelectValue = granteesSelect?.state?.value;
+    const granteesSelectOptions = granteesSelect?.props?.options;
     const payload: EntitySharePayload = {
       selected_grantee_capabilities: entityShareState.selectedGranteeCapabilities,
     };
 
-    return EntityShareActions.update(entityGRN, payload).then(onClose);
+    if (garnteesSelectValue) {
+      const selectedOption = granteesSelectOptions?.find((option) => option.value === garnteesSelectValue);
+
+      if (!selectedOption) {
+        throw Error(`Can't find ${garnteesSelectValue} in grantees select options on save`);
+      }
+
+      // eslint-disable-next-line no-alert
+      if (!window.confirm(`"${selectedOption.label}" got selected but was never added as a collaborator. Do you want to continue anyway?`)) {
+        setDisableSubmit(false);
+
+        return;
+      }
+    }
+
+    EntityShareActions.update(entityGRN, payload).then(() => {
+      setDisableSubmit(true);
+      onClose();
+    });
```

Comment:
I tried to add a more complex test but this is not possible with due to the mocking and usage of the `EntityShareStore`. We will be able to handle the mocking better when we use the return value of our actions inside the `EntityShareModal` instead of the state in the `EntityShareState`